src/index.ts

```diff
@@ -1,19 +1,649 @@
+
+    const { data, ...options } = dataItem;
+
+    const signer: Signer = {
+      publicKey: publicKeyBuffer,
+      signatureType: 1,
+      signatureLength: 512,
+      ownerLength: 512,
+      sign: this.api.getSignerSignFn(sub),
+      // Note we don't provide `verify` as it's not used anyway:
+      // verify: null,
+    };
+
+    const opts: DataItemCreateOptions = {
+      ...options,
+      tags: [...(options.tags || []), ...ANALYTICS_TAGS],
+    };
+
+    const dataItemInstance = createData(data, signer, opts);
+
+    // DataItem.sign() sets the DataItem's `id` property and returns its `rawId`:
+    await dataItemInstance.sign(signer);
+
+    // TODO: ArConnects types the return type as ArrayBuffer, but in the example this goes straight into the DataItem constructor, which only accepts Buffer.
+    // new DataItem(dataItemInstance.getRaw().buffer);
+    // return dataItemInstance.getRaw().buffer;
+    return dataItemInstance.getRaw();
+  }
+
+  /**
+   * Sign the given message. This function assumes (and requires) a user is logged in.
+   * @param message The message to sign.
+   * @returns The signed version of the message.
+   */
+  async signMessage(
+    data: string | BinaryDataType,
+    options?: SignMessageOptions,
+  ): Promise<Uint8Array> {
+    const sub = this.auth0.getCachedUserSub();
+
+    if (!sub) throw new Error("Missing cached user.");
+
+    const hashAlgorithm = options?.hashAlgorithm || "SHA-256";
+
+    const hashArrayBuffer = await this.crypto.subtle.digest(
+      hashAlgorithm,
+      binaryDataTypeOrStringToBinaryDataType(data),
+    );
+
+    const signature = await this.api.sign(hashArrayBuffer, sub);
+
+    return stringToUint8Array(signature);
+  }
+
+  /**
+   * Verify the given message. This function assumes (and requires) a user is logged in.
+   * @param signature The signature to verify.
+   * @returns The signed version of the message.
+   */
+  async verifyMessage(
+    data: string | BinaryDataType,
+    signature: string | BinaryDataType,
+    publicKey: string,
+    options: SignMessageOptions = { hashAlgorithm: "SHA-256" },
+  ): Promise<boolean> {
+    const hashAlgorithm = options?.hashAlgorithm || "SHA-256";
+
+    const hashArrayBuffer = await this.crypto.subtle.digest(
+      hashAlgorithm,
+      binaryDataTypeOrStringToBinaryDataType(data),
+    );
+
+    const publicJWK: JsonWebKey = {
+      e: "AQAB",
+      ext: true,
+      kty: "RSA",
+      n: publicKey,
+    };
+
+    const cryptoKey = await this.crypto.subtle.importKey(
+      "jwk",
+      publicJWK,
+      {
+        name: "RSA-PSS",
+        hash: options.hashAlgorithm,
+      },
+      false,
+      ["verify"],
+    );
+
+    const result = await this.crypto.subtle.verify(
+      { name: "RSA-PSS", saltLength: 32 },
+      cryptoKey,
+      binaryDataTypeOrStringToBinaryDataType(signature),
+      hashArrayBuffer,
+    );
+
+    return result;
+  }
+
+  async privateHash(
```

Comment:
@7i7o Added this missing method using the implementation found in the other functions above. I guess the hash is produced in the same way in all cases.